### PR TITLE
[fix](Nereids): order of project's logical properties is different with that of project expression

### PR DESCRIPTION
**Pull request by keanji-x**
# Proposed changes

1. Add `fromRule` to mark the rule that create this group expression for debuging
2. fix a bug that the order of project's logical properties is different with that of project expression

For example, in some rules like `LOGICAL_SEMI_JOIN_LOGICAL_JOIN_TRANSPOSE_PROJECT`, we create project with a different order, e.g.,
```
project(a, b) -> project(b, a) 
```
However, the logical properties of the group are not changed. Therefore, we get
```
project(b, a) with logical properties (a, b)
```
It causes a binding error when translating physical project, where we use `logicalproperties` to generate `tupleDesc` but calculate the actual output by projects expression

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 1]
```diff
@@ -57,6 +57,8 @@ public class GroupExpression {
 
     private long estOutputRowCount = -1;
 
+    private Rule fromRule;
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java:61]: done

**Comment by jackwener** [comment]
run buildp0

**Review by morrySnow — APPROVED** [review]

**Pull request merged**